src/vs/editor/common/viewModel/characterHardWrappingLineMapper.ts

```diff
@@ -114,6 +114,11 @@ export class CharacterHardWrappingLineMapperFactory implements ILineMapperFactor
 					wrappedTextIndent = '';
 					wrappedTextIndentVisibleColumn = 0;
 				}
+
+				// Markdown wrapped text indent
+				if (strings.startsWith(lineText, `${wrappedTextIndent}- `)) {
```

Comment:
This may be the right place to fix this problem but this code applies to every programming language. We can't hardcode a markdown specific solution here